components/brave_rewards/core/contribution/contribution.cc

```diff
@@ -628,46 +696,53 @@ void Contribution::RetryUnblindedContribution(
 }
 
 void Contribution::Result(const mojom::Result result,
+                          const std::string& queue_id,
                           const std::string& contribution_id) {
-  if (result == mojom::Result::RETRY_SHORT) {
-    SetRetryTimer(contribution_id, base::Seconds(5));
-    return;
-  }
-
-  if (result == mojom::Result::RETRY_LONG) {
-    SetRetryTimer(contribution_id, base::Minutes(5));
-    return;
-  }
-
-  if (result == mojom::Result::RETRY) {
-    SetRetryTimer(contribution_id, util::GetRandomizedDelay(base::Seconds(45)));
-    return;
-  }
-
-  auto get_callback = std::bind(&Contribution::OnResult, this, _1, result);
+  auto get_callback =
+      std::bind(&Contribution::OnResult, this, _1, result, queue_id);
 
   ledger_->database()->GetContributionInfo(contribution_id, get_callback);
 }
 
 void Contribution::OnResult(mojom::ContributionInfoPtr contribution,
-                            const mojom::Result result) {
+                            const mojom::Result result,
+                            const std::string& queue_id) {
+  // Notify any waiting callbacks that the contribution request has either
+  // succeeded or failed. Note that if the contribution was "split" between
+  // multiple funding sources, then the callback will only receive the
+  // completion status for the first completed transaction. In addition, a
+  // successful completion status only indicates that the transaction was
+  // successfully initiated. There may be a significant delay before the
+  // transaction is completed by the provider.
+  OnContributionRequestCompleted(queue_id, result == mojom::Result::LEDGER_OK);
+
   if (!contribution) {
     BLOG(0, "Contribution is null");
     return;
   }
 
+  if (result == mojom::Result::RETRY_SHORT) {
+    SetRetryTimer(contribution->contribution_id, base::Seconds(5));
+    return;
+  }
+
   if (result == mojom::Result::RETRY_LONG) {
     if (contribution->processor == mojom::ContributionProcessor::BRAVE_TOKENS) {
       SetRetryTimer(contribution->contribution_id,
                     util::GetRandomizedDelay(base::Seconds(45)));
     } else {
-      SetRetryTimer(contribution->contribution_id,
-                    util::GetRandomizedDelay(base::Seconds(450)));
+      SetRetryTimer(contribution->contribution_id, base::Minutes(5));
```

Comment:
This doesn't need randomization?

Reply:
The old version of this line (with the randomized delay) was actually in a dead branch. We'll need to make sure that the constant delay is sufficient though.

Reply:
I think this is consistent with existing (and desired) behavior. For vBAT AC, the delay will be random, but for external wallet providers it will be the 5 minutes. @szilardszaloki can you verify please?

Reply:
There's a reason for the `base::Minutes(5)` delay. Going to Slack DM to discuss.